packages/interaction-kit/src/definitions/application-commands.ts

```diff
@@ -30,17 +33,21 @@ export enum ApplicationCommandType {
 	MESSAGE = 3,
 }
 
-/** @link https://discord.com/developers/docs/interactions/slash-commands#application-command-object-application-command-option-structure */
+/** @link https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure */
 export type ApplicationCommandOption = {
 	type: ApplicationCommandOptionType;
 	name: string;
 	description: string;
 	required?: boolean;
 	choices?: ApplicationCommandOptionChoice[];
 	options?: ApplicationCommandOption[];
+	channel_types?: ChannelType[];
+	min_value?: number;
+	max_value?: number;
+	autocomplete?: boolean;
```

Comment:
These can't be on every type of option, I think we should separate them out.

Furthermore, its a pain in the ass, but I'd love if TS could prevent you from  setting...say choices and autocomplete

Reply:
Split them apart (https://github.com/IanMitchell/interaction-kit/pull/101/commits/3f1c7005184b9374a33f4b52b62e9429eb472d65) as discussed in discord